_search-plugins/knn/performance-tuning.md

```diff
@@ -98,6 +98,57 @@ In OpenSearch 2.15 or later, you can further improve indexing speed and reduce d
 This is an expert-level setting. Disabling the `_recovery_source` may lead to failures during peer-to-peer recovery. Before disabling the `_recovery_source`, check with your OpenSearch cluster admin to determine whether your cluster performs regular flushes before starting the peer-to-peer recovery of shards before disabling the `_recovery_source`.  
 {: .warning}
 
+### (Expert-level) Build vector data structures on demand
+
+This approach is recommended only for workloads that involve a single initial bulk upload and will be used exclusively for search after force merging to a single segment.
+
+During indexing, vector search builds a specialized data structure for a `knn_vector` field to enable efficient approximate k-NN search. However, these structures are rebuilt from scratch during [force merge]({{site.url}}{{site.baseurl}}/api-reference/index-apis/force-merge/) for k-NN indexes. To optimize indexing speed, follow these steps:
```

Comment:
"from scratch" is idiomatic. Can we just remove it or replace it with something like "totally rebuilt"? Additionally, following that, "during [force merge] operations on k-NN indexes"?

Reply:
Reworded.